state-sync/state-sync-v2/data-streaming-service/src/data_stream.rs

```diff
@@ -301,31 +300,27 @@ impl<T: DiemDataClient + Send + Clone + 'static> DataStream<T> {
         // Process any ready data responses
         for _ in 0..MAX_CONCURRENT_REQUESTS {
             if let Some(pending_response) = self.pop_pending_response_queue() {
-                let pending_response = pending_response.lock();
+                let mut pending_response = pending_response.lock();
                 let client_response = pending_response
                     .client_response
-                    .as_ref()
+                    .take()
```

Comment:
part of this PR refactors the code so that responses are passed by value rather than by reference. we need owned handles to the responses since the context can't be cloned (since it contains a type-erased `Box<dyn ResponseContext>`). this is definitely one small downside when compared w/ the response_id approach, which you can easily copy around, but I think it also enforces some stronger discipline, so we don't unnecessarily copy these potentially chunky response objects.

here we're just pulling out the response (so we can own it) rather than taking a reference to it.